### PR TITLE
Augment `db-analyser` backingstore CLI with LMDB `mapsize` option.

### PR DESCRIPTION
**Pull request by jorisdral**
Changes:
* Update the flags/options in `db-analyser` for picking UTxO HD backing stores.
* Add an option to set the `mapsize` of the LMDB backing store in case `db-analyser` uses the LMDB backing store.
* Elaborated on the flags/options in `db-analyser/Documentation.md`.

Why:
* Adding the option to set the LMDB `mapsize` allows us to profile the LMDB backing store WRT to changes in the `mapsize` value.

Bugfixes:
* `--store-ledger` was flushing inconsistent state to the backing store.

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - [x] New tests are added if needed and existing tests are updated
    - [x] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [x] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [x] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

**Review by Jasagredo — APPROVED** [review]
Looks good! awesome catch!!

**Review by dnadales — APPROVED** [review]

File: ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs [hunk 1]
```diff
@@ -319,8 +320,30 @@ storeLedgerStateAt slotNo AnalysisEnv { db, backing, initLedger, cfg, ledgerDbFS
       Right (ldb, _) -> do
         -- The replay performed above leaves the DbChangelog at @tip - k@
         -- blocks. We use this flush to push all the remaining differences into
-        -- the backing store.
-        OnDisk.flush backingStore $ ledgerDbChangelog ldb
+        -- the backing store. First, we must update the @DbChangeLog@ such that
+        -- it reflects the fact that we are flushing all remaining differences.
+
```
Comment (dnadales) [inline ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs:325]: Maybe expand a bit on this.

File: ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {-# LANGUAGE BangPatterns        #-}
 {-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE GADTs               #-}
```
[the remark this answers is not in the document]
Comment (jorisdral) [reply ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs:3]: We don't, I removed it. Thanks!

File: ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs (hunk 1 above)
Comment (jorisdral) [reply ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs:334]: My latest commit should expand on this, and include the TODO.

**Comment by jorisdral** [comment]
Merging this when CI is green

**Pull request merged**